netbox/circuits/models/circuits.py

```diff
@@ -276,6 +282,43 @@ class CircuitTermination(
         blank=True
     )
 
+    # Cached associations to enable efficient filtering
+    _provider_network = models.ForeignKey(
+        to='circuits.ProviderNetwork',
+        on_delete=models.PROTECT,
+        related_name='circuit_terminations',
+        blank=True,
+        null=True
+    )
+    _location = models.ForeignKey(
+        to='dcim.Location',
+        on_delete=models.CASCADE,
+        related_name='_circuit_terminations',
+        blank=True,
+        null=True
+    )
+    _site = models.ForeignKey(
+        to='dcim.Site',
+        on_delete=models.CASCADE,
+        related_name='_circuit_terminations',
+        blank=True,
+        null=True
+    )
+    _region = models.ForeignKey(
+        to='dcim.Region',
+        on_delete=models.CASCADE,
+        related_name='_circuit_terminations',
+        blank=True,
+        null=True
+    )
+    _sitegroup = models.ForeignKey(
```

Comment:
Something I wasn't sure about when working on #6414 was whether to go with `sitegroup` or `site_group`. Whichever naming scheme we choose, we should be consistent with e.g. `provider_network` above.